README.md

```diff
@@ -899,6 +904,87 @@ gradlew aemInstall -Paem.package.skipDownloadName=false
 ```
 Only matters when Vault properties file is customized then that property could be used to eliminate conflicts.
 
+## Contributing
+
+### Basics
+
+* Work on fork of this repository - In the top-right corner of this page, click Fork.
+* All your changes should be created from and merged into **develop** branch.
+* If you want your changes to be included in upstream release Create Pull Request from your fork **develop** branch into upstream **develop** branch
+    * Before you create Pull Request make sure that plugin is working with **AEM [Single|Multi]-Project Example** projects
+
+### Testing local version of plugin
+
+After you have done some changes make sure that local version plugin works with one of **AEM Project Example**:
+* [AEM Single-Project Example](https://github.com/Cognifide/gradle-aem-single#quickstart) 
+* [AEM Multi-Project Example](https://github.com/Cognifide/gradle-aem-multi#quickstart) 
+
+by publishing your developed version of **Gradle AEM Plugin** to a local maven.
+
+
+Clone into one of **AEM Project Example** projects.
+
+Add local maven *mavenLocal()* on top of repositories in **build.gradle** file:
+
+```groovy
+pluginManagement {
+    repositories {
+        mavenLocal()
+        jcenter()
+        maven { url "https://plugins.gradle.org/m2/" }
+        maven { url  "http://dl.bintray.com/cognifide/maven-public" }
+        maven { url  "https://dl.bintray.com/neva-dev/maven-public" }
+    }
+    ...
+}
+```
+
+Change version of *com.cognifide.gradle:aem-plugin* in **build.gradle** file:
+
+```groovy
+pluginManagement {
+       resolutionStrategy {
+        eachPlugin {
+        ...
+                useModule('com.cognifide.gradle:aem-plugin:4.0.x')
+        ...
+        }
+    }
+}
+```
+Go back into **Gradle AEM Plugin** project.
+
+Update plugin version in build.gradle to the one you have configured in **AEM Project Example**.
+
+```groovy
+group 'com.cognifide.gradle'
+version '4.0.x'
+description = 'Gradle AEM Plugin'
+defaultTasks = ['clean', 'build', 'publishToMavenLocal']
+``` 
+
+Build **Gradle AEM Plugin**
+
+```bash
+sh gradlew
+```
+
+by default it will run publish to local maven task.
+
+Now **AEM Project Example** will use your local version of **Gradle AEM Plugin**.
+
+### Debug
+
+Execute **AEM Project Example** build with debug options (make sure it is using **local** version of **Gradle AEM Plugin**), it will suspend.
+
+```bash
+sh grdlew -Dorg.gradle.debug=true --no-daemon
+```
+
+Attach **Gradle AEM Plugin** debugger on port 5005.
```

Comment:
avoid concrete naming, use `example project`, `plugin project` instead

Attach debugger in plugin project opened e.g in Intellij by creating Run Configuration for Remote Java Application at port 5005 (default).